Cibuildwheel

Zuzu-Typ opened the pull request:
Updated CI (#43)

mattgwwalker commented:
This looks great to me.  Thanks for setting it up.  Would you like to merge this?

What do these lines achieve?  I see you've added them to all the files I added recently.
```
import os

os.chdir(os.path.dirname(__file__))
```

If you need to test the PyPI packaging then I'm all for it.  How about we go with version 0.7a1?  I'm sure we'll need to go through a few iterations before we get it right.

Cheers,

Matthew

Zuzu-Typ commented:
> What do these lines achieve? I see you've added them to all the files I added recently. I understand that they change the directory to the current file's directory, but _why_ was that necessary?
> 
> ```
> import os
> 
> os.chdir(os.path.dirname(__file__))
> ```

The reason I added these lines is so that you wouldn't be required to run the tests from the `/tests` directory as you previously were, as for some reason pytest calls them from wherever it was executed from - therefore the tests fail as the requirements can't be found.
This also has the advantage that you can now simply call `pytest` from the root directory and it will automatically find the tests and run them.

Zuzu-Typ commented:
@mattgwwalker Okay, theoretically, the way this should work is that we would pusblish a new version here on GitHub and CI should automatically produce the necessary wheels and publish them here and on PyPI - if it works correctly that is.
